release_automation.sh

```diff
@@ -75,12 +75,17 @@ else
     ohai "Confirmed that Aztec Libraries are set to release versions. Proceeding..."
 fi
 
-## Check current branch is develop, trunk, or release/* branch
+# Check if current HEAD is develop or trunk branch
 CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
-if [[ ! "$CURRENT_BRANCH" =~ ^develop$|^trunk$|^release/.* ]]; then
-    warn "Releases should generally only be based on 'develop', 'trunk', or an earlier release branch."
-    warn "You are currently on the '$CURRENT_BRANCH' branch."
-    confirm_to_proceed "Are you sure you want to create a release branch from the '$CURRENT_BRANCH' branch?"
+if [[ ! "$CURRENT_BRANCH" =~ ^develop$|^trunk$ ]]; then
+
+    # Check if current HEAD is a release tag
+    CURRENT_TAG=$(git tag --points-at HEAD)
+    if [[ ! "$CURRENT_TAG" =~ v[0-9]+\.[0-9]+\.[0-9]+ ]]; then
+        warn "Releases should generally only be based on 'develop', 'trunk', or a release tag."
+        warn "Gutenberg-Mobile is currently on the '$CURRENT_BRANCH' branch and '$CURRENT_TAG' tag."
+        confirm_to_proceed "Are you sure you want to create a release branch from here?"
```

Comment:
Maybe this actually isn't so important since there is also the prompt that says what the current version is and that will alert you if you're not on the tag you intend.

Reply:
I agree, I don't think it's so important to duplicate the details of the current branch and tag here as it's already laid out in the warning. In fact, I think the extra details would unnecessarily clutter the messages, which I think are pretty clear as they are:

<img width="632" alt="Screenshot 2022-02-14 at 13 55 35" src="https://user-images.githubusercontent.com/2998162/153877335-8a169511-ce4d-429a-b5cd-f83ff5995036.png">

A nice addition could be to detect when the current tag is an empty string and to then not include a mention of it at all in the warning, but I don't think it's overly important either, as the messaging is still clear.

Reply:
I kind of like having the `''` tag as an indicator that there is no tag on the current code. I feel like that could be useful in cases where someone is intending to branch off of a tag but, for example, maybe they checked out a similarly named branch instead. Admittedly, there are ways to make that clearer than just an empty string indicator, but I'm not sure it's worth the effort to make this message more robust. Happy to revisit it if you'd like though.

Reply:
That reasoning makes sense, I agree it's likely not worth the effort to create something more robust when it's clear and understandable as is 👍